tests/base/aligned_vector_04.cc

```diff
@@ -31,14 +31,14 @@
 class FunctionBase
 {
 public:
-  ~FunctionBase()
+  virtual ~FunctionBase()
   {}
 
   virtual void
   do_test() = 0;
 };
 
-class Function
+class Function : public FunctionBase
```

Comment:
This is curious -- the test previously did not use `FunctionBase` at all. You could have just removed that class. So I'm curious what error/warning you get here?

Reply:
```
1034: In file included from /mnt/data/darndt/Sources/dealii-clang-6.0.0/tests/base/aligned_vector_04.cc:26:
1034: In file included from /mnt/data/darndt/Sources/dealii-clang-6.0.0/include/deal.II/base/table.h:21:
1034: /mnt/data/darndt/Sources/dealii-clang-6.0.0/include/deal.II/base/aligned_vector.h:891:11: error: destructor called on non-final 'Function' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
1034:           (--_end_data)->~T();
1034:           ^
1034: /mnt/data/darndt/Sources/dealii-clang-6.0.0/include/deal.II/base/aligned_vector.h:704:3: note: in instantiation of member function 'dealii::AlignedVector<Function>::clear' requested here
1034:   clear();
1034:   ^
1034: /mnt/data/darndt/Sources/dealii-clang-6.0.0/tests/base/aligned_vector_04.cc:75:27: note: in instantiation of member function 'dealii::AlignedVector<Function>::~AlignedVector' requested here
1034:   AlignedVector<Function> vec;
1034:                           ^
1034: /mnt/data/darndt/Sources/dealii-clang-6.0.0/include/deal.II/base/aligned_vector.h:891:27: note: qualify call to silence this warning
1034:           (--_end_data)->~T();
1034:                           ^
1034: /mnt/data/darndt/Sources/dealii-clang-6.0.0/include/deal.II/base/aligned_vector.h:803:9: error: destructor called on non-final 'Function' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
1034:         (--_end_data)->~T();
1034:         ^
1034: /mnt/data/darndt/Sources/dealii-clang-6.0.0/tests/base/aligned_vector_04.cc:76:7: note: in instantiation of member function 'dealii::AlignedVector<Function>::resize' requested here
1034:   vec.resize(2);
1034:       ^
1034: /mnt/data/darndt/Sources/dealii-clang-6.0.0/include/deal.II/base/aligned_vector.h:803:25: note: qualify call to silence this warning
1034:         (--_end_data)->~T();
1034:                         ^
1034: /mnt/data/darndt/Sources/dealii-clang-6.0.0/include/deal.II/base/aligned_vector.h:486:13: error: destructor called on non-final 'Function' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
1034:             source_[i].~T();
1034:             ^
1034: /mnt/data/darndt/Sources/dealii-clang-6.0.0/include/deal.II/base/aligned_vector.h:870:11: note: in instantiation of member function 'dealii::internal::AlignedVectorMove<Function>::apply_to_subrange' requested here
1034:           dealii::internal::AlignedVectorMove<T>(
1034:           ^
1034: /mnt/data/darndt/Sources/dealii-clang-6.0.0/include/deal.II/base/aligned_vector.h:805:3: note: in instantiation of member function 'dealii::AlignedVector<Function>::reserve' requested here
1034:   reserve(size_in);
1034:   ^
1034: /mnt/data/darndt/Sources/dealii-clang-6.0.0/tests/base/aligned_vector_04.cc:76:7: note: in instantiation of member function 'dealii::AlignedVector<Function>::resize' requested here
1034:   vec.resize(2);
1034:       ^
1034: /mnt/data/darndt/Sources/dealii-clang-6.0.0/include/deal.II/base/aligned_vector.h:486:25: note: qualify call to silence this warning
1034:             source_[i].~T();
1034:                         ^
```
I just decided to turn the classes into something that looks reasonable.

Reply:
Ah. So it would have been enough to declare the destructor of the `Function` class as `virtual` and remove `FunctionBase` altogether?

It's the testsuite, so I'm not overly concerned about any of this...